src/lib/builders/color-picker/create.ts

```diff
@@ -0,0 +1,898 @@
+        },
+        action: (node: HTMLButtonElement) => {
+            const rect = node.getBoundingClientRect();
+
+            alphaPickerDims.set({
+                height: rect.height,
+                width: rect.width,
+            });
+
+            const unsubEvents = executeCallbacks(
+                addMeltEventListener(node, 'mousedown', () => {
+                    alphaDragging = true;
+                }),
+                addMeltEventListener(node, 'mouseup', () => {
+                    alphaDragging = false;
+                }),
+                addMeltEventListener(node, 'keydown', (e) => {
+                    const { key } = e;
+                    const keys = ['ArrowLeft', 'ArrowRight'];
+
+                    if (!keys.includes(key)) return;
+
+                    e.preventDefault();
+
+                    const alpha = get(alphaValue);
+
+                    // Move the picker button and restrict movement to within the canvas.
+                    if (key === 'ArrowRight' && alpha + 1 <= 100) {
+                        alphaValue.set(alpha + 1);
+                    } else if (key === 'ArrowLeft' && alpha - 1 >= 0) {
+                        alphaValue.set(alpha - 1);
+                    }
+                })
+            );
+
+            return {
+                destroy() {
+                    unsubEvents();
+                },
+            }
+        }
+    });
+
+    const eyeDropper = builder(name('eye-dropper'), {
+        returned: () => {
+            return {
+                'aria-label': 'An eye dropper button, allowing you to select any color on the screen.'
+            }
+        },
+        action: (node: HTMLButtonElement) => {
+
+            const eyeWindow = <EyeDropperWindow>window;
+
+            if (eyeWindow.EyeDropper) {
+                eye = new eyeWindow.EyeDropper();
+            }
+
+            const unsubEvents = executeCallbacks(
+                addMeltEventListener(node, 'click', () => {
+                    if (!eye) return;
+
+                    eye
+                        .open()
+                        .then((result) => updateOnColorInput(result.sRGBHex));
+                        // .catch((e) => console.log(e));
+                })
+            );
+
+            return {
+                destroy() {
+                    unsubEvents();
+                }
+            }
+        }
+    });
+
+    const hexInput = builder(name('hex-input'), {
+        action: (node: HTMLInputElement) => {
+
+            const unsubEvents = executeCallbacks(
+                addMeltEventListener(node, 'keyup', () => {
+                    const { value } = node;
+
+                    if (!isValidHex(value)) return;
+
+                    color.set(value);
+                })
+            );
+
+            return {
+                destroy() {
+                    unsubEvents();
+                }
+            }
+        }
+    });
+
+    // Helper functions
+    function RGBtoHex(rgb: ColorRGB) {
```

Comment:
The ones that are not store dependent can be outside of the builder fn (same file, or in `internal/helpers`) for better organization